api/src/test/java/org/pmiops/workbench/exfiltration/EgressRemediationServiceTest.java

```diff
@@ -0,0 +1,408 @@
+    workbenchConfig.egressAlertRemediationPolicy = suspendXMinutesOnXIncidentsPolicy();
+
+    // events 1, 2 should be merged but 3 should not
+    saveOldEvents(Duration.ofMinutes(300), Duration.ofMinutes(250), Duration.ofMinutes(200));
+    egressRemediationService.remediateEgressEvent(saveNewEvent());
+
+    // 1 merged old incident, 1 standalone, and the active event
+    assertComputeSuspended(Duration.ofMinutes(3));
+  }
+
+  @Test
+  public void testRemediateEgressEvent_incidentsExcludeFalsePostives() {
+    workbenchConfig.egressAlertRemediationPolicy = suspendXMinutesOnXIncidentsPolicy();
+
+    saveOldEvents(
+        oldEvent(Duration.ofHours(3)).setStatus(EgressEventStatus.VERIFIED_FALSE_POSITIVE));
+
+    egressRemediationService.remediateEgressEvent(saveNewEvent());
+
+    // Only the active event is considered
+    assertComputeSuspended(Duration.ofMinutes(1));
+  }
+
+  @Test
+  public void testRemediateEgressEvent_suspendCompute() {
+    workbenchConfig.egressAlertRemediationPolicy.escalations =
+        ImmutableList.of(suspendComputeAfter(1, Duration.ofMinutes(1)));
+
+    long eventId = saveNewEvent();
+    egressRemediationService.remediateEgressEvent(eventId);
+
+    assertThat(getDbUser().getDisabled()).isFalse();
+    assertComputeSuspended(Duration.ofMinutes(1));
+
+    DbEgressEvent event = egressEventDao.findById(eventId).get();
+    assertThat(event.getStatus()).isEqualTo(EgressEventStatus.REMEDIATED);
+  }
+
+  @Test
+  public void testRemediateEgressEvent_disableUser() {
+    workbenchConfig.egressAlertRemediationPolicy.escalations =
+        ImmutableList.of(disableUserAfter(1));
+
+    long eventId = saveNewEvent();
+    egressRemediationService.remediateEgressEvent(eventId);
+
+    assertThat(getDbUser().getDisabled()).isTrue();
+    assertComputeNotSuspended();
+    // The disable action also stops the user's runtimes
+    verify(mockLeonardoNotebooksClient).stopAllUserRuntimesAsService(USER_EMAIL);
+
+    DbEgressEvent event = egressEventDao.findById(eventId).get();
+    assertThat(event.getStatus()).isEqualTo(EgressEventStatus.REMEDIATED);
+  }
+
+  private void saveOldEvents(Duration... ages) {
+    saveOldEvents(
+        Arrays.stream(ages)
+            .map(this::oldEvent)
+            .collect(Collectors.toList())
+            .toArray(new DbEgressEvent[] {}));
+  }
+
+  private void saveOldEvents(DbEgressEvent... events) {
+    for (DbEgressEvent target : events) {
+      saveEventWithCreationTimeWorkaround(target);
+    }
+  }
+
+  private long saveNewEvent() {
+    return saveNewEvent(newEvent());
+  }
+
+  private long saveNewEvent(DbEgressEvent e) {
+    return saveEventWithCreationTimeWorkaround(e).getEgressEventId();
+  }
+
+  private DbEgressEvent saveEventWithCreationTimeWorkaround(DbEgressEvent target) {
+    Timestamp targetCreation = target.getCreationTime();
+
+    // Hack: insert the event, then update it. Apply the desired creation timestamp on the update.
```

Comment:
FTR I tried https://stackoverflow.com/a/63530418 but could not get the default `@CreationTimestamp` result to change.